modules/integration/src/test/scala/scala/cli/integration/CompileTestDefinitions.scala

```diff
@@ -251,4 +251,19 @@ abstract class CompileTestDefinitions(val scalaVersionOpt: Option[String])
           )
         )
     }
+
+  test("JVM options only for JVM platform") {
+    val inputs = TestInputs(
+      Seq(os.rel / "Main.scala" -> "// using java-opt -Xss1g")
+    )
+    inputs.fromRoot { root =>
+      val res = os.proc(TestUtil.cli, "compile", extraOptions, "--native", ".").call(
```

Comment:
If you agree with my comment above, maybe we should keep this test but check that it doesn't error. And add one specifying the native platform via a using directive, that should error out.